### PR TITLE
Hamburger Menu

### LINKED ISSUE #237: Hamburger menu on mobile devices
**Is your feature request related to a problem? Please describe.**
On mobile devices the navigation isn't optimal. 
localhost_8081_course_2021_W_FKvDB(iPhone 12 Pro)

**Describe the solution you'd like**
A button on the top right of the header which opens a hamburger menu (or something similiar).

### PR DESCRIPTION
**Pull request by MatthiasReumann**
Closes #237 (Add hamburger menu on mobile devices)

### CODE REVIEW

**Review by alexanderstephan — CHANGES_REQUESTED** [review]
Looks good to me codewise, design wise I also really like this!

When a played around with it, I noticed a couple minor things:
I noticed that somehow there is a lot free space in the menu in my browser, see the image blow. Resolution is 637x874 in this example.


Maybe we can reduce the right margin? We could also think about putting the dark theme and GitHub icon directly below the other menu items, I guess.

I also think that in light mode, the menu items could be almost black instead of grey. I have a hard time reading the text here.

I noticed that the "Login" spacing is different on my Honor 6x Android 7 Phone on Firefox. I can't reproduce it on my Desktop though. So not sure, if this easily fixable or even a problem on my end.

File: web/template/header.gohtml [hunk 1]
```diff
@@ -1,56 +1,123 @@
...
+                        <span class="ml-3">Start</span>
+                    </a>
+                </div>
+                {{- /*gotype: TUM-Live/tools.TUMLiveContext*/ -}}
+                {{if not .User}}
+                    <div class="pl-5 py-3 dark:border-secondary">
+                        <a href="/login"
+                           class="font-semibold text-gray-400">
+                            <i class="fas fa-sign-in-alt w-8"></i>
+                            <span>Login</span>
```
Comment (joschahenningsen) [inline web/template/header.gohtml]:
```suggestion
                            <span class="ml-3">Login</span>
```

Comment (joschahenningsen) [reply web/template/header.gohtml]: This should fix @alexanderstephan s observation

**Comment by joschahenningsen** [comment]
Other than what @alexanderstephan pointed out, this looks great! I agree that there is quite a bit of space here, but I don't mind.

**Comment by MatthiasReumann** [comment]
> Looks good to me codewise, design wise I also really like this!
> 
> When a played around with it, I noticed a couple minor things: I noticed that somehow there is a lot free space in the menu in my browser, see the image blow. Resolution is 637x874 in this example.
> 
> 
> 
> Maybe we can reduce the right margin? We could also think about putting the dark theme and GitHub icon directly below the other menu items, I guess.
> I also think that in light mode, the menu items could be almost black instead of grey. I have a hard time reading the text here.
> 
> I noticed that the "Login" spacing is different on my Honor 6x Android 7 Phone on Firefox. I can't reproduce it on my Desktop though. So not sure, if this easily fixable or even a problem on my end. 

- Fixed the Login bug (Thanks @joschahenningsen!) 
- Changed the dark-mode button to a toggle thingy 
- Moved the GitHub and toggle button up
- Light mode text black

**Review by joschahenningsen — APPROVED** [review]
Looks super clean :fire:

**Review by alexanderstephan — APPROVED** [review]
Looks awesome, I also really like the addition of the dark mode toggle! :fire: Maybe this means we should close https://github.com/joschahenningsen/TUM-Live/pull/134?

**Pull request merged**